### ISSUE TITLE
Infinite loading of the world when used with TerrainControl + BiomeBundle

### ISSUE THREAD

**Issue by hhiru** [Shinoow/AbyssalCraft#189]
I made a thread here with some info: https://www.reddit.com/r/biomebundle/comments/5rfg1n/infinite_world_loading/

For some reason I can't generate a world with TerrainControl if AbyssalCraft is present in the modlist. I have to manually close java causing a crash.

Here's the crash file:
[crash-2017-02-01_15.14.59-server.txt](https://github.com/Shinoow/AbyssalCraft/files/745065/crash-2017-02-01_15.14.59-server.txt)

**Comment by Shinoow** [comment]
https://www.reddit.com/r/feedthebeast/comments/5ln7rf/biome_bundle_1102_forge_finally_released/dbxpvcu/
You haven't properly configured TerrainControl. Since BiomeBundle doesn't work with mods out of the box, some config stuff needs to be tweaked if you run it alongside other mods that adds biomes to the Overworld.

**Comment by hhiru** [comment]
Since TC uses vanilla biomes to make their biomes I have to edit AbyssalCraft biome IDs to make it work with TC, right?

**Comment by Shinoow** [comment]
Something like that. As Pitman said in the beginning of that thread, Worldconfig.ini in the TC config folder is where you have to change stuff in order for it to function properly.

**Comment by hhiru** [comment]
I'm not sure of what biome ID is conflicting with yours tho. there are probably hundreds of IDs in that file :/

**Comment by Shinoow** [comment]
Since biome IDs switched to being internal (rather than a set number you could make configurable) in 1.9, trial and error is pretty much the way to go.
The vanilla biome IDs can be found at http://minecraft.gamepedia.com/Biome#Biome_IDs

**Comment by hhiru** [comment]
I still don't know what biomeID TC should not use in order to prevent conflicts. what ID is your mod using?

**Comment by Shinoow** [comment]
I've never used the mod myself, so I can't contribute to anything to that. Since the biome IDs are dynamically assigned now, the theoretical range (provided no other mod present registers their biomes before) would be 40 - 52, where the Overworld biomes would be 40, 46, 47, 48, 49, 50.

**Comment by hhiru** [comment]
The IDs used in that range by TC are these: Frozen Forest:44, Autumnal Woods Hills:47, Autumnal Woods M:48, Autumnal Woods:49, Tropical Savanna:50

To fix the problem I looked at the bottom of the list and chose IDs greater than the last ID that is SnowyPlainsEdge:635 so in this case the first 5 biomes in the list becomes: Frozen Forest:636, Autumnal Woods Hills:637, Autumnal Woods M:638, Autumnal Woods:639, Tropical Savanna:640 

There is no need to move the biomes at the end of the list, I just had to change the IDs and on the next modload TC moved the first 5 biomes at the end of the list since their IDs were bigger than the last ( SnowyPlainsEdge:635 ). I'm wondering if I can do the same with Biomes O' Plenty now. Thank you for the input!
